### PR TITLE
Add Mojo execution tests and 'sourceGenerationFormat' property

### PR DESCRIPTION
**Pull request by makubi**
* sourceGenerationFormat maps to SourceFormat of avrohugger
* Tests for Mojo executions

### CODE REVIEW

File: src/main/java/at/makubi/maven/plugin/avrohugger/Defaults.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+package at.makubi.maven.plugin.avrohugger;
+
+interface Defaults {
+    /** Relative to ${basedir} */
+    String relativeSourceDirectory = "src/main/avro";
+    /** Relative to ${project.build.directory} */
+    String relativeOutputDirectory = "generated-sources/avro";
+    String recursive = "false";
+    String limitedNumberOfFieldsInCaseClasses = "false";
+    String sourceGenerationFormat = "SPECIFIC_RECORD";
```
Comment (mariussoutier) [inline src/main/java/at/makubi/maven/plugin/avrohugger/Defaults.java:26]: Can you use the `SourceGenerationFormat` enum here?

Comment (mariussoutier) [inline src/main/java/at/makubi/maven/plugin/avrohugger/Defaults.java:26]: I'd also use a boolean or enum for the two boolean fields, better documentation, less stringly typing.

**Review by mariussoutier — CHANGES_REQUESTED** [review]

File: src/main/java/at/makubi/maven/plugin/avrohugger/Defaults.java (hunk 1 above)
Comment (makubi) [reply src/main/java/at/makubi/maven/plugin/avrohugger/Defaults.java:26]: I would like to, but the Maven property defaults have to be constant String values.

See https://github.com/makubi/avrohugger-maven-plugin/blob/583138ba70081950e01905c64b2162b1799a45af/src/main/java/at/makubi/maven/plugin/avrohugger/GeneratorMojo.java#L42.

**Review by mariussoutier — APPROVED** [review]

File: src/main/java/at/makubi/maven/plugin/avrohugger/Defaults.java (hunk 1 above)
Comment (mariussoutier) [reply src/main/java/at/makubi/maven/plugin/avrohugger/Defaults.java:26]: Too bad, but ok for the time being.

**Pull request merged**